### PR TITLE
Make RELEASE_NOTES.md a bit less outdated

### PR DESCRIPTION
**Pull request by SvyatoslavScherbina**

### CODE REVIEW

**Comment by SvyatoslavScherbina** [comment]
I also need some help with "Debugging" section: mentioning IDE debugging support might be useful.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -55,28 +55,34 @@ Produced programs are fully self-sufficient and do not need JVM or other runtime
...
-  During the first run it will download all the external dependencies, such as LLVM.
-
-To see the list of available flags, run `kotlinc -h`.
-
-For documentation on C interoperability stubs see [INTEROP.md](https://github.com/JetBrains/kotlin-native/blob/master/INTEROP.md).
+  
+The most complete experience with Kotlin/Native can be achieved by using
+[Gradle](https://kotlinlang.org/docs/tutorials/native/using-gradle.html),
+[IntelliJ IDEA](https://kotlinlang.org/docs/tutorials/native/using-intellij-idea.html) or
+[Android Studio with KMM plugin if you target iOS](https://kotlinlang.org/docs/mobile/create-first-app.html).
```
Comment (ilmat192) [inline RELEASE_NOTES.md:75]: May be mark as a link only the "Android Studio with KMM plugin" part? Like
```suggestion
[Android Studio with KMM plugin](https://kotlinlang.org/docs/mobile/create-first-app.html)  if you target iOS.
```

**Comment by sbogolepov** [comment]
Let's remove
>  Adding support for other target platforms shouldn't be too hard, if LLVM support is available.

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -55,28 +55,34 @@ Produced programs are fully self-sufficient and do not need JVM or other runtime
...
 
-The language and library version supported by this release match Kotlin 1.4.
+The language and library version supported by this release match Kotlin 1.5.
 However, there are certain limitations, see section [Known Limitations](#limitations).
 
  Currently _Kotlin/Native_ uses reference counting based memory management scheme with a cycle
 collection algorithm. Multiple threads could be used, but objects must be explicitly transferred
-between threads, and same object couldn't be accessed by two threads concurrently.
+between threads, and same object couldn't be accessed by two threads concurrently unless it is frozen.
+See the relevant [documentation](https://kotlinlang.org/docs/reference/native/concurrency.html).
```
Comment (sbogolepov) [inline RELEASE_NOTES.md:64]: May be mention blog-post about new GC/MM?

**Comment by sbogolepov** [comment]
> WebAssembly (-target wasm32) target, hosted on Linux, Windows or macOS. Webassembly support is experimental and **could** be discontinued in further releases.

Could -> will?

**Comment by sbogolepov** [comment]
> Experimental support is available on Linux and macOS hosts (requires Wine)

I suggest to drop this part.

File: RELEASE_NOTES.md (hunk 2 above)
Comment (SvyatoslavScherbina) [reply RELEASE_NOTES.md:60]: Done.

**Comment by SvyatoslavScherbina** [comment]
@sbogolepov 
> Let's remove
> 
> > To enable experimental targets Kotlin/Native must be recompiled with org.jetbrains.kotlin.native.experimentalTargets Gradle property set.
> 
> > Adding support for other target platforms shouldn't be too hard, if LLVM support is available.

Done.

> > WebAssembly (-target wasm32) target, hosted on Linux, Windows or macOS. Webassembly support is experimental and **could** be discontinued in further releases.
> 
> Could -> will?

Too early for this.

> > Experimental support is available on Linux and macOS hosts (requires Wine)
> 
> I suggest to drop this part.

Done.

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -25,15 +25,15 @@ the following platforms:
  * Ubuntu Linux x86-64 (14.04, 16.04 and later), other Linux flavours may work as well, host and target
    (`-target linux_x64`, default on Linux hosts, hosted on Linux, Windows and macOS).
  * Microsoft Windows x86-64 (tested on Windows 7 and Windows 10), host and target (`-target mingw_x64`,
-   default on Windows hosts). Experimental support is available on Linux and macOS hosts (requires Wine).
+   default on Windows hosts).
  * Microsoft Windows x86-32 cross-compiled target (`-target mingw_x86`), hosted on Windows.
    Experimental support is available on Linux and macOS hosts (requires Wine).
```
Comment (sbogolepov) [inline RELEASE_NOTES.md:30]:
```suggestion
```

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -44,39 +44,44 @@ the following platforms:
  * Experimental support for Zephyr RTOS (`-target zephyr_stm32f4_disco`) is available on macOS, Linux
```
Comment (sbogolepov) [inline RELEASE_NOTES.md:44]: We can stop mentioning this one as well.
```suggestion
```

Comment (SvyatoslavScherbina) [reply RELEASE_NOTES.md:44]: Why do you think so? Sounds somewhat premature.

Comment (sbogolepov) [reply RELEASE_NOTES.md:43]: 1. We don't even know is it working :)
2. Even if it does, UX is just terrible.
I'm not suggesting to drop Zephyr support from compiler. Just to stop attracting users with the feature that is barely usable.

Comment (SvyatoslavScherbina) [reply RELEASE_NOTES.md:43]: Made the mention more explicit instead of removing it.

**Review by sbogolepov — APPROVED** [review]

**Review by ilmat192 — APPROVED** [review]

File: RELEASE_NOTES.md [hunk 5]
```diff
@@ -55,28 +51,37 @@ Produced programs are fully self-sufficient and do not need JVM or other runtime
...
+We are going to lift these multithreading restrictions, which involves implementing a new memory manager.
+More details are available in
+["Kotlin/Native Memory Management Roadmap"](https://blog.jetbrains.com/kotlin/2020/07/kotlin-native-memory-management-roadmap/).
 
-_Kotlin/Native_ provides efficient interoperability with libraries written in C or Objective-C, and supports
-automatic generation of Kotlin bindings from a C/Objective-C header file.
-See the samples coming with the distribution.
+_Kotlin/Native_ provides efficient bidirectional interoperability with C and Objective-C.
+See the [samples](https://github.com/JetBrains/kotlin-native/tree/master/samples)
+and the [tutorials](https://kotlinlang.org/docs/tutorials/).
```
Comment (p7nov) [inline RELEASE_NOTES.md:67]: We'll have to return to doc links once we completely migrate the docs to WebHelp, i.e. next week (we hope 🤞).
We can change this one to https://kotlinlang.org/docs/native-c-interop.html

**Review by p7nov — APPROVED** [review]

File: RELEASE_NOTES.md (hunk 5 above)
Comment (SvyatoslavScherbina) [reply RELEASE_NOTES.md:67]: Ok, makes sense. Thanks!

**Pull request merged**